packages/valory/skills/market_creation_manager_abci/rounds.py

```diff
@@ -305,6 +339,114 @@ def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
         return None
 
 
+class CollectProposedMarketsRound(CollectSameUntilThresholdRound):
+    """CollectProposedMarketsRound"""
+
+    ERROR_PAYLOAD = "ERROR_PAYLOAD"
+    MAX_RETRIES_PAYLOAD = "MAX_RETRIES_PAYLOAD"
+    MAX_APPROVED_MARKETS_REACHED_PAYLOAD = "MAX_APPROVED_MARKETS_REACHED_PAYLOAD"
+    SKIP_MARKET_APPROVAL_PAYLOAD = "SKIP_MARKET_APPROVAL_PAYLOAD"
+
+    payload_class = CollectProposedMarketsPayload
+    synchronized_data_class = SynchronizedData
+
+    def end_block(self) -> Optional[Tuple[BaseSynchronizedData, Event]]:
+        """Process the end of the block."""
+        if self.threshold_reached:
+            if self.most_voted_payload == self.ERROR_PAYLOAD:
+                proposed_markets_api_retries = cast(
+                    SynchronizedData, self.synchronized_data
+                ).proposed_markets_api_retries
+                synchronized_data = self.synchronized_data.update(
+                    synchronized_data_class=SynchronizedData,
+                    **{
+                        get_name(
+                            SynchronizedData.proposed_markets_api_retries
+                        ): proposed_markets_api_retries
+                        + 1,
+                    },
+                )
```

Comment:
This should be updated on the behaviour side.

Reply:
Agree, but the framework does not allow update synchronized data on the behaviour.

Reply:
Values should be stored as received by the payload. You can pass it from the behaviour via the payload. Ideally, the round should not override the `end_block` method.

Reply:
Fixed